controllers/machinedeployment_controller.go

```diff
@@ -203,7 +206,27 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, cluster *cl
 		if d.Spec.Strategy.RollingUpdate == nil {
 			return ctrl.Result{}, errors.Errorf("missing MachineDeployment settings for strategy type: %s", d.Spec.Strategy.Type)
 		}
-		return ctrl.Result{}, r.rolloutRolling(ctx, d, msList)
+
+		// If we are not in a rolling update operation, rollout after if we can.
+		newMachineSet, oldMachineSets, err := r.getAllMachineSetsAndSyncRevision(ctx, d, msList, true)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		allMachineSets := append(oldMachineSets, newMachineSet)
+		oldMachinesCount := mdutil.GetReplicaCountForMachineSets(oldMachineSets)
+		// If there's no oldMachinesCount we assume we are not in a rolling update operation.
+		if oldMachinesCount == 0 {
+			if err := r.rolloutAfter(ctx, d, newMachineSet); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			// Update status and requeue.
+			return ctrl.Result{}, r.syncDeploymentStatus(allMachineSets, newMachineSet, d)
+		}
+
+		if err := r.rolloutRolling(ctx, d, msList); err != nil {
```

Comment:
Previously we always returned here, also when err was nil, now we don't. Was that intentional?